File: backend/chat-service/src/main/java/com/example/chatservice/chat/dto/ChatRoomLastReadChatDto.java
package com.example.chatservice.chat.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatRoomLastReadChatDto {

    private String roomId;
    private Long lastSendChatId;
    private Long lastReadChatId;
    private Long unreadChatCount;

    public void updateUnreadChatCount(Long unreadChatCount) {
        this.unreadChatCount = unreadChatCount;
    }

    public static ChatRoomLastReadChatDto createChatRoomLastReadChatDto(String roomId, Long lastSendChatId, Long lastReadChatId, Long unreadChatCount) {
        return ChatRoomLastReadChatDto.builder()
                .roomId(roomId)
                .lastSendChatId(lastSendChatId)
                .lastReadChatId(lastReadChatId)
                .unreadChatCount(unreadChatCount)
                .build();
    }
}

File: backend/chat-service/src/main/kotlin/com/example/chatservice/chat/dto/ChatRoomLastReadChatDto.kt
package com.example.chatservice.chat.dto

data class ChatRoomLastReadChatDto(
    var roomId: String? = null,
    var lastSendChatId: Long? = null,
    var lastReadChatId: Long? = null,
    var unreadChatCount: Long? = null,
) {

    fun updateUnreadChatCount(unreadChatCount: Long?) {
        this.unreadChatCount = unreadChatCount
    }

    companion object {
        @JvmStatic
        fun createChatRoomLastReadChatDto(roomId: String?, lastSendChatId: Long?, lastReadChatId: Long?, unreadChatCount: Long?) =
            ChatRoomLastReadChatDto(roomId, lastSendChatId, lastReadChatId, unreadChatCount)
    }
}
